fe/fe-core/src/main/java/org/apache/doris/analysis/UpdateStmt.java

```diff
@@ -211,12 +212,21 @@ private void analyzeSetExprs(Analyzer analyzer) throws AnalysisException {
                 && !hasVariant) {
             isPartialUpdate = true;
         }
+        Optional<Column> seqColInTable = Optional.empty();
+        OlapTable olapTable = (OlapTable) targetTable;
+        if (olapTable.hasSequenceCol() && olapTable.getSequenceMapCol() != null) {
+            seqColInTable = olapTable.getFullSchema().stream()
+                    .filter(col -> col.getName().equalsIgnoreCase(olapTable.getSequenceMapCol())).findFirst();
+        }
         for (Column column : targetTable.getColumns()) {
             Expr expr = new SlotRef(targetTableRef.getAliasAsName(), column.getName());
             boolean existInExpr = false;
             for (BinaryPredicate setExpr : setExprs) {
                 Expr lhs = setExpr.getChild(0);
-                if (((SlotRef) lhs).getColumn().equals(column)) {
+                Column exprColumn = ((SlotRef) lhs).getColumn();
+                if (exprColumn.equals(column) || (olapTable.hasSequenceCol()
```

Comment:
Add comment here

Reply:
@zhannngchen already change the code, take a look plz.